java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -704,6 +707,148 @@ public static CompletableFuture<Long> arrlen(
+     * assert res.equals(gs("{\"a\": 42, \"b\": 33}"));
+     * }</pre>
+     */
+    public static CompletableFuture<GlideString> arrpop(
+            @NonNull BaseClient client, @NonNull GlideString key) {
+        return executeCommand(client, new GlideString[] {gs(JSON_ARRPOP), key});
+    }
+
+    /**
+     * Pops the last element from the array stored in the root of the JSON document stored at <code>
+     * key</code>.
+     *
+     * @param client The Valkey GLIDE client to execute the command.
+     * @param key The <code>key</code> of the JSON document.
+     * @param options Options including the path and optional index, see {@link JsonArrPopOptions}.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>):<br>
+     *           Returns an array with a strings for every possible path, representing the popped JSON
+     *           values, or <code>null</code> for JSON values matching the path that are not an array
+     *           or an empty array. If a value is not an array, its corresponding return value is
+     *           <code>"null"</code>.
+     *       <li>For legacy path (<code>path</code> doesn't start with <code>$</code>):<br>
+     *           Returns a string representing the popped JSON value, or <code>null</code> if the
+     *           array at <code>path</code> is empty. If multiple paths are matched, the value from
+     *           the first matching array that is not empty is returned. If <code>path</code> doesn't
+     *           exist or the value at <code>path</code> is not an array, an error is raised.
+     *     </ul>
+     *     If <code>key</code> doesn't exist, an error is raised.
+     * @example
+     *     <pre>{@code
+     * String doc = "{\"a\": [1, 2, true], \"b\": {\"a\": [3, 4, [\"value\", 3, false], 5], \"c\": {\"a\": 42}}}";
+     * Json.set(client, "doc", "$", doc).get();
+     * var res = Json.arrpop(client, "doc", JsonArrPopOptions.builder().path("$.a").index(1).build()).get();
+     * assert res.equals("2"); // Pop second element from array at path `$.a`
+     *
+     * res = Json.arrpop(client, "doc", JsonArrPopOptions.build().path("$..a").build()).get();
+     * assert Arrays.equals(new Object[] { "true", "5", null }); / Pop last elements from all arrays matching path `..a`
+     *
+     * res = Json.arrpop(client, "doc", JsonArrPopOptions.build().path("..a").build()).get();
+     * assert res.equals("1"); // First match popped (from array at path `$.a`)
+     *
+     * Json.set(client, "doc", "$", "[[], [\"a\"], [\"a\", \"b\", \"c\"]]").get();
+     * res = Json.arrpop(client, "doc", JsonArrPopOptions.builder().path(".").index(-1).build()).get());
+     * assert res.equals("[\"a\", \"b\", \"c\"]"); // Pop last elements at path `.`
+     * }</pre>
+     */
+    public static CompletableFuture<Object> arrpop(
+            @NonNull BaseClient client, @NonNull String key, @NonNull JsonArrPopOptions options) {
```

Comment:
I'm not a big fan how path is required and part of the options.  
This makes is different from all the other commands in that path isn't required in method signature, but hidden within the options.

Reply:
I suppose the advantage is that we don't need to have just path and path+index as separate method signatures...

Reply:
Updated